File: Kotlin/app/src/main/java/com/example/kotlin_fa/MainActivity.kt
package com.example.kotlin_fa

import android.graphics.*
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import com.example.kotlin_fa.databinding.ActivityMainBinding
import java.io.File
import java.io.InputStream


class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        val navController = findNavController(R.id.nav_host_fragment_content_main)
        appBarConfiguration = AppBarConfiguration(navController.graph)
        setupActionBarWithNavController(navController, appBarConfiguration)

        /*
        binding.fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }
        */
        val prime = findViewById<Button>(R.id.prime);
        prime.setOnClickListener {
            val numb: MutableList<Int> = mutableListOf(1)
            var yes: Boolean = false
            for (i in 1..200000) {
                yes = false;
                for (x in 2..i)
                {
                    if(i%x==0)
                    {
                        yes = false;
                        break;
                    }else
                    {
                        yes = true;
                    }
                }
                if(yes==true)
                {
                    numb.add(i);
                    yes = false;
                }
            }
            Toast.makeText(this,"prime finished",Toast.LENGTH_SHORT).show();
        }
        val pic = findViewById<Button>(R.id.pic);
        pic.setOnClickListener {
                // get input stream
            val ims: InputStream = assets.open("test.png")
            val bitmap = BitmapFactory.decodeStream(ims)
            var x: Int = 0
            var y: Int = 0
            var res: String = ""
            //Toast.makeText(this,bitmap.getPixel(0,0).toString(),Toast.LENGTH_SHORT).show();
            repeat(bitmap.width)
            {
                //Toast.makeText(this,x.toString(),Toast.LENGTH_SHORT).show();
                repeat(bitmap.width)
                {

                    if(bitmap.getPixel(x,y).toString()=="")
                    {
                        res = (bitmap.getPixel(x,y)*5).toString()
                    }
                    y = y.inc()
                }
                y = 0
                x = x.inc()
            }
            Toast.makeText(this,"pic is done",Toast.LENGTH_SHORT).show();
        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        return navController.navigateUp(appBarConfiguration)
                || super.onSupportNavigateUp()
    }
}

File: Xamarin/app2/app2/MainActivity.cs
using System;
using System.IO;
using System.Drawing;
using System.Collections.Generic;
using Xamarin.Essentials;
using Xamarin.Android.Net;
using Android.App;
using Android.Content.PM;
using SkiaSharp;
using Android.Runtime;
using Android.Graphics.Drawables;
using Android.OS;
using Android.Content.Res;
using Android.Views;
using AndroidX.AppCompat.Widget;
using AndroidX.AppCompat.App;
using Google.Android.Material.FloatingActionButton;
using Google.Android.Material.Snackbar;

namespace App2
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            Toolbar toolbar = FindViewById<Toolbar>(Resource.Id.toolbar);
            SetSupportActionBar(toolbar);

            FloatingActionButton fab = FindViewById<FloatingActionButton>(Resource.Id.fab);
            fab.Click += FabOnClick;
            FloatingActionButton fab2 = FindViewById<FloatingActionButton>(Resource.Id.fab2);
            fab2.Click += FabOnClick2;
        }

        public override bool OnCreateOptionsMenu(IMenu menu)
        {
            MenuInflater.Inflate(Resource.Menu.menu_main, menu);
            return true;
        }

        public override bool OnOptionsItemSelected(IMenuItem item)
        {
            int id = item.ItemId;
            if (id == Resource.Id.action_settings)
            {
                return true;
            }

            return base.OnOptionsItemSelected(item);
        }

        private void FabOnClick(object sender, EventArgs eventArgs)
        {
            int no1 = (Int32)(DateTime.Now.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
            int beg = 0;
            int end = 200000;
            bool yes = false;
            List<int> prime = new List<int>();
            int div = Math.Abs(end - beg);
            for (int i = 0; i < div; i++)
            {
                yes = false;
                for (int j = 2; j < i; j++)
                {
                    if (i % j == 0)
                    {
                        yes = false;
                        break;
                    }
                    else
                    {
                        yes = true;
                    }
                }
                if (yes == true)
                {
                    prime.Add(i);
                    yes = false;
                }

            }
            int no2 = (Int32)(DateTime.Now.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
            View view = (View) sender;
            Snackbar.Make(view, (no2 - no1).ToString(), Snackbar.LengthLong)
                .SetAction("Action", (View.IOnClickListener)null).Show();
            System.Diagnostics.Debug.WriteLine(prime.ToString());
        }

        private void FabOnClick2(object sender, EventArgs eventArgs)
        {
            AssetManager assets = Android.App.Application.Context.Assets;
            int no1 = (Int32)(DateTime.Now.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
            string res = "";
            using (var stream = this.Assets.Open(@"test.png"))
            { 
                Android.Graphics.Bitmap bm = Android.Graphics.BitmapFactory.DecodeStream(stream);
                Console.WriteLine("done");
                for (int i = 0;i < bm.Width; i++)
                {
                    for (int y = 0; y < bm.Height; y++)
                    {
                        res = (bm.GetPixel(i,y)*5).ToString();
                    }
                }
            }
            int no2 = (Int32)(DateTime.Now.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
            View view = (View)sender;
            Snackbar.Make(view, (no2 - no1).ToString(), Snackbar.LengthLong)
                .SetAction("Action", (View.IOnClickListener)null).Show();
        }

        

        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, [GeneratedEnum] Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }
        
	}
}
